### ISSUE TITLE
fine-tuning with additional masked lm loss, and masked lm loss diverged

### ISSUE THREAD

**Issue by yyht** [google-research/bert#306]
![image](https://user-images.githubusercontent.com/14133687/50417047-daa97f00-085e-11e9-9584-55efbdbd15b9.png)

I had tried to do fine-tuning with additional masked lm loss, however, during training phrase, the masked lm loss diverged with various corresponding masked lm loss weight.
Could anyone provide some solutions?

**Comment by xwzhong** [comment]
how many steps between storing? if is small, it is normal.

**Comment by yyht** [comment]
storing model after each epoch(6w data and batch_size=32)

**Comment by xwzhong** [comment]
you may try lower init learning rate, in my experiment, i use 1e-5(for sentence length 50, 64) or 1e-4(for sentence length 128)

**Comment by yyht** [comment]
thanks, I found a bug in my code, and after fixing it, my experiment seems work well.
![image](https://user-images.githubusercontent.com/14133687/50437133-8d431580-0923-11e9-8218-6398a3a4efde.png)

**Comment by GEORGE5961** [comment]
What's your bug? I'm likely to come across a similar bug.

**Comment by yyht** [comment]
i used scope to collect parameters but forgot to add params of masked lm modules

**Comment by qiunian711** [comment]
> i used scope to collect parameters but forgot to add params of masked lm modules
@yyht 
Could anyone provide some solutions?
I use my Chinese corpus to do fine-tune on bert.  I segment the corpus, then filter out the low-frequency words to generate a vocab.txt, and then use bert's weight to initialize the network.
But mask_lm_loss to 3 has not been able to continue, next_sentence_loss is not bad 0.12, do you know what is the reason?
Is this because I don't have a good dictionary? I don't know if you can give some advice?

您好，可以请教您一些问题吗？
就是我现在也在用自己的中文语料在bert上做fine-tune，词典是自己做的分词然后过滤掉低频词生成的，只是用了bert的权重来初始化网络。但是mask_lm_loss到3一直降不下去，next_sentence_loss还不错0.12，您知道这个是什么原因吗？这个会是我词典没有处理好的原因吗？不知道您是否可以给出一些建议？
